src/controllers/wallet/tx-template/tx-template.controller.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const { SendTransaction } = require('@hathor/wallet-lib');
+const { mapTxReturn, runSendTransaction } = require('../../../helpers/tx.helper');
+const { lockSendTx } = require('../../../helpers/lock.helper');
+const { parametersValidation } = require('../../../helpers/validations.helper');
+const { cantSendTxErrorMessage } = require('../../../helpers/constants');
+const { DEFAULT_PIN } = require('../../../constants');
+
+/**
+ * @typedef {import('@hathor/wallet-lib').HathorWallet} HathorWallet
+ * @typedef {import('winston').Logger} Logger
+ * @typedef {{ walletId: string, wallet: HathorWallet, logger: Logger }} HathorRequestExtras
+ * @typedef {import('express').Request} ExpressRequest
+ * @typedef {ExpressRequest & HathorRequestExtras} Request
+ * @typedef {import('express').Response} Response
+ */
+
+/**
+ * Build a transaction from the tx-template and push to the network.
+ * @param {Request} req
+ * @param {Response} res
+ */
+async function runTemplate(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const unlock = lockSendTx(req.walletId);
+  if (unlock === null) {
+    res.send({ success: false, error: cantSendTxErrorMessage });
```

Comment:
thought(non-blocking): These error handling lines were not tested. Maybe in the future we could have a helper function to test all routes with blocking features in a formatted way.